### PR TITLE
[oakdex-pokedex] Added type defs for oakdex-pokedex

### PR DESCRIPTION
**Pull request by jalyna**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/oakdex-pokedex/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,351 @@
+// Type definitions for oakdex-pokedex 0.4
+// Project: https://github.com/jalyna/oakdex-pokedex
+// Definitions by: Jalyna Schroeder <https://github.com/jalyna>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface Translations {
+    de: string,
```
Comment (SimonSchick) [inline types/oakdex-pokedex/index.d.ts]: Nit: `,` should be `;` in all interfaces.

File: types/oakdex-pokedex/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+export function allRegions(conditions?: Conditions): Region[]
+
+export function allEggGroups(conditions?: Conditions): EggGroup[]
+
+export function allGenerations(conditions?: Conditions): Generation[]
+
+export function allNatures(conditions?: Conditions): Nature[]
+
+export {}
```
Comment (SimonSchick) [inline types/oakdex-pokedex/index.d.ts]: NIt: is this needed?

**Review by SimonSchick — APPROVED** [review]
Looks ok overall, some minor style issue.

Also I noticed you initially added the types to your package but then removed them from the published bundle, did you run into any issues we might be able to help with?

**Comment by jalyna** [comment]
@SimonSchick Thanks for your feedback, all applied :heart:

**Review by SimonSchick — APPROVED** [review]

File: types/oakdex-pokedex/tslint.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "semicolon": [
```
Comment (sandersn) [inline types/oakdex-pokedex/tslint.json]: Can you please remove these overrides and fix the resulting lint?

**Review by sandersn — CHANGES_REQUESTED** [review]

**Review by sandersn — APPROVED** [review]

**Pull request merged**